packages/block-library/src/navigation-submenu/index.php

```diff
@@ -289,3 +289,26 @@ function register_block_core_navigation_submenu() {
 	);
 }
 add_action( 'init', 'register_block_core_navigation_submenu' );
+
+/**
+ * Disables the display of tabs for the Navigation Submenu block.
+ *
+ * @param array $settings Default editor settings.
+ * @return array Filtered editor settings.
+ */
+function gutenberg_disable_tabs_for_navigation_submenu_block( $settings ) {
```

Comment:
is it worth having a shared file for functions that both blocks use?

Reply:
TBH these blocks are so similar it would also make sense to share JS code as well. Is there a precedent for this? 

Can we make one block canonical and import the code into the other block from the other's directory?

Reply:
> is it worth having a shared file for functions that both blocks use?

I did look at something along these lines in the example diff below. Ultimately, it doesn't seem worth the change, the potential clutter in core etc. Merging the two blocks almost into a block variation type situation sounds like a better option down the road.



<details>
<summary>Example Diff</summary>

```diff
diff --git a/lib/experimental/editor-settings.php b/lib/experimental/editor-settings.php
index 862a1bfe2b..d36dce2d14 100644
--- a/lib/experimental/editor-settings.php
+++ b/lib/experimental/editor-settings.php
@@ -72,6 +72,29 @@ function gutenberg_initialize_editor( $editor_name, $editor_script_handle, $sett
 	);
 }
 
+/**
+ * Flags the specified block to disable block inspector tabs.
+ *
+ * @param string $block_name Block name.
+ * @param array  $settings   Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function gutenberg_disable_block_inspector_tabs_for_block( $block_name, $settings ) {
+	$current_tab_settings = _wp_array_get(
+		$settings,
+		array( '__experimentalBlockInspectorTabs' ),
+		array()
+	);
+
+	$settings['__experimentalBlockInspectorTabs'] = array_merge(
+		$current_tab_settings,
+		array( $block_name => false )
+	);
+
+	return $settings;
+}
+
 /**
  * Sets a global JS variable used to trigger the availability of each Gutenberg Experiment.
  */
diff --git a/packages/block-library/src/navigation-link/index.php b/packages/block-library/src/navigation-link/index.php
index 8853d49f6e..a558c49e83 100644
--- a/packages/block-library/src/navigation-link/index.php
+++ b/packages/block-library/src/navigation-link/index.php
@@ -361,18 +361,7 @@ add_action( 'init', 'register_block_core_navigation_link' );
  * @return array Filtered editor settings.
  */
 function gutenberg_disable_tabs_for_navigation_link_block( $settings ) {
-	$current_tab_settings = _wp_array_get(
-		$settings,
-		array( '__experimentalBlockInspectorTabs' ),
-		array()
-	);
-
-	$settings['__experimentalBlockInspectorTabs'] = array_merge(
-		$current_tab_settings,
-		array( 'core/navigation-link' => false )
-	);
-
-	return $settings;
+	return gutenberg_disable_block_inspector_tabs_for_block( 'core/navigation-link', $settings );
 }
 
 add_filter( 'block_editor_settings_all', 'gutenberg_disable_tabs_for_navigation_link_block' );
diff --git a/packages/block-library/src/navigation-submenu/index.php b/packages/block-library/src/navigation-submenu/index.php
index 99c86493d5..25de8de0e3 100644
--- a/packages/block-library/src/navigation-submenu/index.php
+++ b/packages/block-library/src/navigation-submenu/index.php
@@ -306,18 +306,7 @@ add_action( 'init', 'register_block_core_navigation_submenu' );
  * @return array Filtered editor settings.
  */
 function gutenberg_disable_tabs_for_navigation_submenu_block( $settings ) {
-	$current_tab_settings = _wp_array_get(
-		$settings,
-		array( '__experimentalBlockInspectorTabs' ),
-		array()
-	);
-
-	$settings['__experimentalBlockInspectorTabs'] = array_merge(
-		$current_tab_settings,
-		array( 'core/navigation-submenu' => false )
-	);
-
-	return $settings;
+	return gutenberg_disable_block_inspector_tabs_for_block( 'core/navigation-submenu', $settings );
 }
 
 add_filter( 'block_editor_settings_all', 'gutenberg_disable_tabs_for_navigation_submenu_block' );

```
</details>